src/catalog/src/durable/impls/persist.rs

```diff
@@ -238,48 +263,69 @@ impl PersistHandle {
         current_upper(&mut self.write_handle).await
     }
 
-    /// Reports if the catalog state has been initialized, and the current upper.
-    async fn is_initialized_inner(&mut self) -> (bool, Timestamp) {
+    /// Generates a timestamp for reading from the catalog that is as fresh as possible, given
+    /// `upper`.
+    fn as_of(&self, upper: Timestamp) -> Timestamp {
+        as_of(&self.read_handle, upper)
+    }
+
+    /// Reports the status of catalog initialization.
+    async fn is_initialized_inner(&mut self) -> InitializedStatus {
         let upper = self.current_upper().await;
-        (upper > 0.into(), upper)
+        if upper == InitializedStatus::ShardUninitialized.upper() {
+            InitializedStatus::ShardUninitialized
+        } else if upper == InitializedStatus::CatalogUninitialized.upper() {
+            InitializedStatus::CatalogUninitialized
+        } else {
+            InitializedStatus::Initialized(upper)
+        }
+    }
+
+    /// Generates a [`Vec<StateUpdate>`] that contain all updates to the catalog
+    /// state.
+    ///
+    /// The output is consolidated and sorted by timestamp in ascending order and the current upper.
+    #[tracing::instrument(level = "debug", skip(self))]
+    async fn current_snapshot(&mut self) -> (Vec<StateUpdate>, Timestamp) {
```

Comment:
These usages of the term "current" should probably be switched to "recent" for consistency.